src/transformers/data/processors/squad.py

```diff
@@ -107,9 +107,13 @@ def squad_convert_example_to_features(
     tok_to_orig_index = []
     orig_to_tok_index = []
     all_doc_tokens = []
+    tokenizer_type = type(tokenizer).__name__.replace("Tokenizer", "").lower()
     for (i, token) in enumerate(example.doc_tokens):
         orig_to_tok_index.append(len(all_doc_tokens))
-        sub_tokens = tokenizer.tokenize(token)
+        if tokenizer_type in ("roberta", "longformer", "bart"):
+            sub_tokens = tokenizer.tokenize(token, add_prefix_space=True)
```

Comment:
Wondering why you choose to use this string-based approach rather than a pure type based `isinstance(tokenizer, (RobertaTokenizer, ..., ....))`

I would say the one based on type is more reliable, wdyt @LysandreJik, @sgugger?

Reply:
Yes, that would be better and cleaner.

Reply:
Sure, happy to change it. Just tried to comply with the already existing `tokenizer_type` that is later used here: 
https://github.com/huggingface/transformers/blob/571c7a11c17bd00ba3e79f4d853cc51428a14e45/src/transformers/data/processors/squad.py#L139

Reply:
Done